src/components/common/ContactPanelProvider/ContactPanelProvider.test.tsx

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import { act, renderHook } from '@testing-library/react-hooks';
+import TestRouter from '__tests__/util/TestRouter';
+import {
+  ContactPanelProvider,
+  getQueryContactId,
+  setQueryContactId,
+  useContactPanel,
+} from './ContactPanelProvider';
+
+const contactId = '00000000-0000-0000-0000-000000000000';
+const newContactId = 'ffffffff-ffff-ffff-ffff-ffffffffffff';
+
+describe('getQueryContactId', () => {
+  it('should return undefined if the contactId is missing', () => {
+    const query = {};
+    expect(getQueryContactId(query, 'contactId')).toBeUndefined();
+  });
+
+  it('should return undefined if the contactId is a string', () => {
+    const query = { contactId };
+    expect(getQueryContactId(query, 'contactId')).toBeUndefined();
+  });
+
+  it('should return undefined if the contactId is an empty array', () => {
+    const query = { contactId: [] };
+    expect(getQueryContactId(query, 'contactId')).toBeUndefined();
+  });
+
+  it('should return undefined if the contactId is not a UUID', () => {
+    const query = { contactId: ['flows'] };
+    expect(getQueryContactId(query, 'contactId')).toBeUndefined();
+  });
+
+  it('should return the contact id if the contactId is a UUID', () => {
+    const query = { contactId: [contactId] };
+    expect(getQueryContactId(query, 'contactId')).toBe(contactId);
+  });
+
+  it('should return the contact id if the contactId is an array ending with a UUID', () => {
+    const query = { contactId: [contactId] };
+    expect(getQueryContactId(query, 'contactId')).toBe(contactId);
+  });
+});
+
+describe('setQueryContactId', () => {
+  it('should remove the existing contactId if contactId is undefined', () => {
+    const query = { contactId: ['flows', contactId], key: 'value' };
+    const result = setQueryContactId(query, 'contactId', undefined);
+    expect(result).toEqual({
+      contactId: ['flows'],
+      key: 'value',
+    });
+  });
+
+  it('should add replace the existing contactId', () => {
+    const query = { contactId: [contactId], key: 'value' };
+    const result = setQueryContactId(query, 'contactId', newContactId);
+    expect(result).toEqual({
+      contactId: [newContactId],
+      key: 'value',
+    });
+  });
+
+  it('should add a new contactId to the array', () => {
+    const query = { contactId: ['flows'], key: 'value' };
+    const result = setQueryContactId(query, 'contactId', newContactId);
+    expect(result).toEqual({
+      contactId: ['flows', newContactId],
+      key: 'value',
+    });
+  });
+
+  it('should handle contactId param not being an array', () => {
+    const query = { contactId: contactId, key: 'value' };
+    const result = setQueryContactId(query, 'contactId', newContactId);
+    expect(result).toEqual({
+      contactId: [newContactId],
+      key: 'value',
+    });
+  });
```

Comment:
Can you add a test for when the person Id is open and closed

Reply:
Since it works in my testing without needing to remove `personId` from the URL, could we do this in a future improvement or not at all to expedite merging this?

Reply:
Sure we can do it later